lib/client.js

```diff
@@ -266,7 +275,11 @@ class Client extends Dispatcher {
         handler = new RedirectHandler(this, maxRedirections, opts, handler)
       }
 
+
       const request = new Request(opts, handler)
+      if (channels.dispatch.hasSubscribers) {
```

Comment:
Since your request/opts objects aren't specifically created for the diagnostics channel, you should be able to just call `publish` without this check. In the event that `diagnostics_channel` is unavailable, this could be a no-op assigned in your catch block near line 34.